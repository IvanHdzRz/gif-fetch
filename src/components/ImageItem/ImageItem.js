import React from 'react'
import Styles from './ImageItem.module.css'

const ImageItem=({id,title,url}) =>{
    return (
        <div className={Styles.imageItem} key={id}> 
            <img src={url}alt={title} />
            <h3>{title}</h3>
        </div>
    )
}



export default ImageItem

