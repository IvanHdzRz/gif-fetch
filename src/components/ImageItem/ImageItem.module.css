.imageItem{
    width: 200px;
    height: 200px;
    background-color: white;
    border-radius: 10px;
    overflow: hidden;
    margin: 10px;
    position: relative;
    box-sizing: border-box;
}
.imageItem>img{
    width: 200px;
    height: 200px;
    object-fit: cover;
    object-position: center;
    
}

.imageItem::before{
    content: '';
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    background: linear-gradient(0deg, rgba(0,0,0,.9) 0%, rgba(253,187,45,0) 100%);
    transition: all .3s;
}

.imageItem>h3{
    font-size: 1rem;
    text-align: center;
    color: white;
    font-family: 'Chilanka', cursive;
    position: absolute;
    bottom: 0;
    right: 50%;
    transform: translateX(50%);
    width: 100%;
    padding: 10px;
    box-sizing: border-box;
    z-index: 100;
}