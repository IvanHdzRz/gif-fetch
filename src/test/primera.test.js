import React from 'react'
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import ImageItem from '../components/ImageItem/ImageItem'

describe('pruebas en  componenteimageItem',()=>{
    test('debe renderizar componente',()=>{
        //preparacion
        const image={title:'example',url:'http://hola.com',id:'12345'}
        //estimulo
        const wrapper=shallow(<ImageItem {...image} />)
        //se espera
        expect(wrapper).toMatchSnapshot();
    })
})
