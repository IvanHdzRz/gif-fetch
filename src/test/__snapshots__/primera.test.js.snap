// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en  componenteimageItem debe renderizar componente 1`] = `
<ImageItem
  id="12345"
  title="example"
  url="http://hola.com"
/>
`;
